components/gmServiceGmail.js

```diff
@@ -120,14 +120,21 @@ gmServiceGmail.prototype = {
     var connection = Components.classes["@gmail-manager-community.github.com/gmanager/connection;1"].createInstance(Components.interfaces.gmIConnection);
     connection.send(this._loginURL, null);
 
-    var serviceURI = {
-      "url": this._loginURL,
-      "data": this._getFirstStepPostData(connection.data, (aPassword || this._password), aContinueData)
-    };
+    var data = connection.data;
+    var postUrl = this._extractActionUrlFromFormData(data);
+    connection.send(postUrl, this._getFirstStepPostData(data), this);
+
+    data = connection.data;
+    postUrl = this._extractActionUrlFromFormData(data);
```

Comment:
Did you miss the `connection.send` between duplication? The second part of code is working on the different response.  

The flow:
1. Get the login page from server - line 121
2. Parse the response and get required data - lines 123-124
3. Post request to the server with account name (1st step login) - line 125
4. Parse the response and get required data - 127-128